packages/wrangler/src/dev.tsx

```diff
@@ -586,8 +579,9 @@ function useEsbuild({
       // esbuild already logs errors to stderr and we don't want to end the process
       // on build errors anyway so this is a no-op error handler
     });
-
-    return stopWatching;
+    return () => {
```

Comment:
React ignores subsequent setups of the cleanup callback, so this needs to be a function on first run. I'm surprised there was no warning thrown here, I kinda remember us working on it back then. Might've been lost at some point.